### ISSUE TITLE
Add "Search unordered items in parallel" example

### ISSUE THREAD

**Issue by budziq** [rust-lang-nursery/rust-cookbook#330]
Showcase [`find_any`](https://docs.rs/rayon/*/rayon/iter/trait.ParallelIterator.html#method.find_any) with [rayon](https://crates.io/crates/rayon) crate.

Please discuss the idea here before starting implementation.
cc https://github.com/rust-lang-nursery/rust-cookbook/issues/272

**Comment by budziq** [comment]
> Do we care about the size of the data we are searching? 

Lets just show the general idea. Making the example data meaningfully parallelizable  is out of scope for such example.

> Since we are searching, maybe we should have examples for find_any, find_first, and find_last in this recipe since their use is similar.

find_last / find_first impose additional restrictions on the iterable and I'd say that their use is more specialized. I'd need a solid and specific use case here to add these.

**Comment by j-haj** [comment]
I have a very simple example [here](https://github.com/j-haj/rayon-search/blob/master/rayon-search/src/main.rs). Everything is pretty straight forward except for the double reference part. Is it worth explaining why there is a `&&x` in the final example?

**Comment by budziq** [comment]
Nice! Actually rayon shares this quirk with standard [sequential find](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find) from std we can always link to it in the example (it's already explained there but we'll see how the textual description unfolds :) )

In terms of the actual example we might go wild here and give some more interesting predicate than just direct equality. 
Hmm I would also go with assert_eq to the actual values

**Comment by j-haj** [comment]
Sounds good! I will add a more complex predicate to the example and submit a PR sometime today or tomorrow hopefully

**Comment by j-haj** [comment]
@budziq I am not sure how wild you were thinking we would get here so I added two different approaches to the example I linked earlier:

1. I added a slightly more complex search predicate (find an even number greater than 6) - a little more complex than straight equality, but not materially

2. Created a `Person` struct that implements the `PartialEq` trait, and search a `Vec<Person>` for a an entry with a specified `first_name` and `age` greater than some given value. This is quite a bit more complex, but might be too complex for the goals of the Cookbook. On the other hand, this example may be closer to "real world" than the integers example
